src/index.js

```diff
@@ -1,52 +1,53 @@
-import {List, Map} from 'immutable';
+import { findIndex } from './array-utils';
 
 export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
 export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
+  if (state instanceof Object) {
+    if (Array.isArray(state.history)) {
+      return state.current;
     }
   }
   return state;
 };
 
-export const preloadState = state => Map({
+export const preloadState = state => ({
   beforeState: undefined,
-  history: List(),
+  history: [],
   current: state
 });
 
 const applyCommit = (state, commitId, reducer) => {
-  const history = state.get('history');
+  const { history } = state;
   // If the action to commit is the first in the queue (most common scenario)
-  if (history.first().meta.optimistic.id === commitId) {
-    const historyWithoutCommit = history.shift();
-    const nextOptimisticIndex = historyWithoutCommit.findIndex(action => action.meta && action.meta.optimistic && !action.meta.optimistic.isNotOptimistic && action.meta.optimistic.id);
+  if (history[0].meta.optimistic.id === commitId) {
+    const historyWithoutCommit = history.slice(1);
+    const nextOptimisticIndex = findIndex(historyWithoutCommit, action => action.meta && action.meta.optimistic && !action.meta.optimistic.isNotOptimistic && action.meta.optimistic.id);
     // If this is the only optimistic item in the queue, we're done!
     if (nextOptimisticIndex === -1) {
-      return state.withMutations(mutState => {
-        mutState
-          .set('history', List())
-          .set('beforeState', undefined)
-      });
+      return {
+        ...state,
+        history: [],
+        beforeState: undefined
+      };
     }
     // Create a new history starting with the next one
-    const newHistory = historyWithoutCommit.skip(nextOptimisticIndex);
+    const newHistory = historyWithoutCommit.slice(nextOptimisticIndex);
     // And run every action up until that next one to get the new beforeState
     const newBeforeState = history.reduce((mutState, action, index) => {
       return index <= nextOptimisticIndex ? reducer(mutState, action) : mutState;
-    }, state.get('beforeState'));
-    return state.withMutations(mutState => {
-      mutState
-        .set('history', newHistory)
-        .set('beforeState', newBeforeState)
-    });
+    }, state.beforeState);
+
+    return {
+      ...state,
+      history: newHistory,
+      beforeState: newBeforeState
+    };
   } else {
     // If the committed action isn't the first in the queue, find out where it is
-    const actionToCommit = history.findEntry(action => action.meta && action.meta.optimistic && action.meta.optimistic.id === commitId);
+    const actionToCommit = history.find(action => action.meta && action.meta.optimistic && action.meta.optimistic.id === commitId);
```

Comment:
Need to change this to use `findIndex` as `Array.prototype.find` isn't supported on IE either.